packages/editor-ui/src/stores/cloudPlan.store.ts

```diff
@@ -4,60 +4,66 @@ import type { CloudPlanState } from '@/Interface';
 import { useRootStore } from '@/stores/n8nRoot.store';
 import { useSettingsStore } from '@/stores/settings.store';
 import { useUsersStore } from '@/stores/users.store';
-import { getCurrentPlan } from '@/api/cloudPlans';
-import { useCloudPlanHelper } from '@/composables/useCloudPlanHelper';
+import { getCurrentPlan, getCurrentUsage } from '@/api/cloudPlans';
+import { DateTime } from 'luxon';
 
 const DEFAULT_STATE: CloudPlanState = {
-	data: {
-		planId: 0,
-		monthlyExecutionsLimit: 0,
-		activeWorkflowsLimit: 0,
-		credentialsLimit: 0,
-		isActive: false,
-		displayName: '',
-		expirationDate: '',
-		metadata: {
-			version: 'v1',
-			group: 'opt-in',
-			slug: 'trial-1',
-		},
-		usage: {
-			executions: 0,
-			activeWorkflows: 0,
-		},
-	},
+	data: null,
+	usage: null,
 };
 
 export const useCloudPlanStore = defineStore('cloudPlan', () => {
 	const rootStore = useRootStore();
 	const settingsStore = useSettingsStore();
 	const usersStore = useUsersStore();
-	const { userIsTrialing: _userIsTrialing } = useCloudPlanHelper();
 
 	const state = reactive<CloudPlanState>(DEFAULT_STATE);
 
 	const setData = (data: CloudPlanState['data']) => {
 		state.data = data;
 	};
 
-	const userIsTrialing = computed(() => _userIsTrialing(state.data?.metadata));
+	const setUsage = (data: CloudPlanState['usage']) => {
+		state.usage = data;
+	};
+
+	const userIsTrialing = computed(() => state.data?.metadata.group === 'trial');
 
 	const currentPlanData = computed(() => state.data);
 
+	const currentUsageData = computed(() => state.usage);
+
+	const trialExpired = computed(
+		() =>
+			state.data?.metadata?.group === 'trial' &&
+			DateTime.now().toMillis() >= DateTime.fromISO(state.data?.expirationDate).toMillis(),
+	);
+
+	const allExecutionsUsed = computed(
+		() => state.usage?.executions === state.data?.monthlyExecutionsLimit,
```

Comment:
unlikely to be an issue if everything is setup correctly.. but would opt for `>=`